E2E test: ACCESS_REFUSED error code path

## Problem

`ResponseCodeEnum::ACCESS_REFUSED` (0x10) exists in the codebase but is never exercised in any E2E test. This error code is returned when a user attempts an operation they don't have permissions for.

## How to trigger

This requires a RabbitMQ user with restricted permissions:
1. Create a user with limited permissions via management API
2. Connect with that user
3. Attempt to create/delete a stream in a vhost where the user has no configure permission

## Expected test

```php
public function testAccessRefusedForRestrictedUser(): void
{
    // Requires Docker setup with a restricted user
    // Could be skipped if management API is not available
    $connection = Connection::create(user: 'restricted', password: 'restricted');

    $this->expectException(\Exception::class);
    $connection->createStream('forbidden-stream');
}
```

## Why it matters

- Validates authorization enforcement at the protocol level
- Important for multi-user deployments
- Ensures error messages are meaningful for debugging

## Notes

- This test may require additional Docker setup (creating restricted users)
- Consider adding it as an optional/skippable test
- Could be combined with vhost access tests (#149)

## Acceptance criteria

- [ ] Test with restricted user → `ACCESS_REFUSED` exception
- [ ] Error message includes the operation that was refused

## CI Pipeline Failed

Pipeline checks failed. Logs saved to `/home/decodo/work/rabbit-stream/.oda/artifacts/152/pipeline-fail.log`.

```
## Failed CI Checks

Failed: E2E Tests

### Run 23751523886

```
E2E Tests	Run E2E tests	﻿2026-03-30T15:00:40.1667697Z ##[group]Run ./vendor/bin/phpunit --testsuite e2e --testdox
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1668196Z [36;1m./vendor/bin/phpunit --testsuite e2e --testdox[0m
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1695913Z shell: /usr/bin/bash -e {0}
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1696147Z env:
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1696328Z   COMPOSER_PROCESS_TIMEOUT: 0
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1696556Z   COMPOSER_NO_INTERACTION: 1
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1696769Z   COMPOSER_NO_AUDIT: 1
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1696959Z   RABBITMQ_HOST: 127.0.0.1
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1697160Z   RABBITMQ_PORT: 5552
E2E Tests	Run E2E tests	2026-03-30T15:00:40.1697339Z ##[endgroup]
E2E Tests	Run E2E tests	2026-03-30T15:00:40.2501614Z PHPUnit 10.5.63 by Sebastian Bergmann and contributors.
E2E Tests	Run E2E tests	2026-03-30T15:00:40.2502143Z 
E2E Tests	Run E2E tests	2026-03-30T15:00:40.2502456Z Runtime:       PHP 8.3.30
E2E Tests	Run E2E tests	2026-03-30T15:00:40.2502985Z Configuration: /home/runner/work/rabbit-stream/rabbit-stream/phpunit.xml
E2E Tests	Run E2E tests	2026-03-30T15:00:40.2503726Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.1888338Z EEE..........................................................SSS. 65 / 79 ( 82%)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9771687Z ..............                                                    79 / 79 (100%)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9772402Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9772586Z Time: 01:36.723, Memory: 10.00 MB
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9772868Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9773372Z Access Refused (CrazyGoat\RabbitStream\Tests\E2E\AccessRefused)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9774280Z  ✘ Create stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9774690Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9775828Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9776968Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9777787Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9778940Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9779614Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9780328Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9781061Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9781752Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9782458Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:77
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9782927Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9783445Z  ✘ Create stream access refused has response code
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9783907Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9784816Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9785851Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9786596Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9787723Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9788387Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9789089Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9790097Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9790748Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9791445Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:97
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9791875Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9792166Z  ✘ Delete stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9792399Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9793060Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9793908Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9794363Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9795012Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9795775Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9796374Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9796954Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9797536Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9798136Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:118
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9798512Z    │
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9798600Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9798817Z Amqp Message Decoder E2E (CrazyGoat\RabbitStream\Tests\E2E\AmqpMessageDecoderE2E)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9799232Z  ✔ Decode simple published message
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9799504Z  ✔ Decode message with properties
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9799795Z  ✔ Decode message with application properties
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800095Z  ✔ Decode multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800244Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800356Z Close (CrazyGoat\RabbitStream\Tests\E2E\Close)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800646Z  ✔ Close command
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800755Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9800901Z Connection (CrazyGoat\RabbitStream\Tests\E2E\Connection)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9801250Z  ✔ Connection create and stream management
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9801537Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9801764Z  ✔ Get metadata
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9801957Z  ✔ Get stream stats
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9802180Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9802337Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9802531Z Connection Handshake (CrazyGoat\RabbitStream\Tests\E2E\ConnectionHandshake)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9802929Z  ✔ Peer properties exchange
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9803371Z  ✔ Sasl handshake
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9803709Z  ✔ Sasl authenticate
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9803913Z  ✔ Full handshake
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9804129Z  ✔ Invalid credentials throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9804268Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9804469Z Connection Resilience (CrazyGoat\RabbitStream\Tests\E2E\ConnectionResilience)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9804911Z  ✔ Operation after socket disconnect
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9805205Z  ✔ Is connected returns false after socket error
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9805553Z  ✔ No resource leaks after socket error
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9805719Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9805848Z Consumer (CrazyGoat\RabbitStream\Tests\E2E\Consumer)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9806157Z  ✔ Read returns empty on timeout
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9806413Z  ✔ Produce and consume with read
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9806669Z  ✔ Read one returns single message
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9806925Z  ✔ Store and query offset
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9807171Z  ✔ Auto commit on close stores last offset
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9807481Z  ✔ No auto commit on close does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9807850Z  ✔ Auto commit on close with no messages does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9808091Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9808235Z Create Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateStream)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9808557Z  ✔ Create stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9808775Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9809210Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9809363Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9809540Z Create Super Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9809915Z  ✔ Create super stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9810152Z  ✔ Create duplicate super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9810322Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9810491Z Declare Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeclarePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9810869Z  ✔ Declare publisher with reference
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9811142Z  ✔ Declare publisher without reference
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9811455Z  ✔ Declare publisher on non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9811693Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9811868Z Delete Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeletePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9812283Z  ✔ Delete publisher after declare
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9812576Z  ✔ Delete non existent publisher throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9812750Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9813110Z Delete Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteStream)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9813788Z  ✔ Delete stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9814011Z  ✔ Delete non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9814195Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9814397Z Delete Super Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9814796Z  ✔ Delete super stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9815081Z  ✔ Delete non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9815279Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9815530Z Exchange Command Versions (CrazyGoat\RabbitStream\Tests\E2E\ExchangeCommandVersions)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9816008Z  ✔ Exchange command versions
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9816175Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9816430Z Named Producer Deduplication (CrazyGoat\RabbitStream\Tests\E2E\NamedProducerDeduplication)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9817001Z  ✔ Named producer deduplication across reconnect
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9817220Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9817453Z Osiris Chunk Parser E2E (CrazyGoat\RabbitStream\Tests\E2E\OsirisChunkParserE2E)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9817939Z  ✔ Parse real chunk from rabbit m q
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9818250Z  ✔ Chunk parser with single message
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9818401Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9818540Z Partitions (CrazyGoat\RabbitStream\Tests\E2E\Partitions)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9818965Z  ✔ Partitions for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9819336Z  ✔ Partitions returns streams for super stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9819539Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9819668Z Producer (CrazyGoat\RabbitStream\Tests\E2E\Producer)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9820005Z  ✔ Send and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9820272Z  ✔ Send batch and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9820553Z  ✔ Get last publishing id
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9820798Z  ✔ Query sequence for named producer
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9821096Z  ✔ Wait for confirms timeout throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9821389Z  ✔ Producer remains usable after timeout
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9821591Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9821877Z Producer Consumer Offset Resume (CrazyGoat\RabbitStream\Tests\E2E\ProducerConsumerOffsetResume)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9822430Z  ✔ Full lifecycle with offset resume
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9822595Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9822730Z Publish (CrazyGoat\RabbitStream\Tests\E2E\Publish)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9823057Z  ✔ Publish single message
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9823564Z  ✔ Publish multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9823711Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9823877Z Publish Error (CrazyGoat\RabbitStream\Tests\E2E\PublishError)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9824331Z  ✔ Publish error callback is invoked when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9824773Z  ✔ Publish error callback is invoked for batch when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9825037Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9825262Z Query Publisher Sequence (CrazyGoat\RabbitStream\Tests\E2E\QueryPublisherSequence)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9825795Z  ✔ Query publisher sequence returns zero for new publisher
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9826216Z  ✔ Query publisher sequence returns last published id
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9826420Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9826547Z Route (CrazyGoat\RabbitStream\Tests\E2E\Route)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9826906Z  ✔ Route for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9827417Z  ✔ Route returns streams for routing key
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9827609Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9827841Z Store Offset Query Offset E2E (CrazyGoat\RabbitStream\Tests\E2E\StoreOffsetQueryOffsetE2E)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9828358Z  ✔ Store and query offset at protocol level
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9828679Z  ✔ Query offset for non existent reference
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9829008Z  ✔ Store offset multiple times last value wins
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9829215Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9829357Z Stream Stats (CrazyGoat\RabbitStream\Tests\E2E\StreamStats)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9829759Z  ✔ Stream stats returns statistics
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9830057Z  ✔ Stream stats for non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9830240Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9830367Z Subscribe (CrazyGoat\RabbitStream\Tests\E2E\Subscribe)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9830716Z  ✔ Subscribe to stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9830949Z  ✔ Subscribe with offset last
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9831398Z  ✔ Subscribe with offset next
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9831703Z  ✔ Subscribe to non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9831993Z  ✔ Duplicate subscription id throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9832172Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9832340Z Unsubscribe (CrazyGoat\RabbitStream\Tests\E2E\Unsubscribe)
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9832692Z  ✔ Unsubscribe from stream
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9832966Z  ✔ Unsubscribe non existent subscription throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9833472Z  ✔ Unsubscribe already unsubscribed throws
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9833766Z 
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9833861Z ERRORS!
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9834085Z Tests: 79, Assertions: 410, Errors: 3, Skipped: 3.
E2E Tests	Run E2E tests	2026-03-30T15:02:16.9897281Z ##[error]Process completed with exit code 2.
```


```

## CI Pipeline Failed

Pipeline checks failed. Logs saved to `/home/decodo/work/rabbit-stream/.oda/artifacts/152/pipeline-fail.log`.

```
## Failed CI Checks

Failed: E2E Tests

### Run 23751808143

```
E2E Tests	Run E2E tests	﻿2026-03-30T15:06:30.2201030Z ##[group]Run ./vendor/bin/phpunit --testsuite e2e --testdox
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2201472Z [36;1m./vendor/bin/phpunit --testsuite e2e --testdox[0m
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2227543Z shell: /usr/bin/bash -e {0}
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2227767Z env:
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2227938Z   COMPOSER_PROCESS_TIMEOUT: 0
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2228162Z   COMPOSER_NO_INTERACTION: 1
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2228365Z   COMPOSER_NO_AUDIT: 1
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2228548Z   RABBITMQ_HOST: 127.0.0.1
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2228739Z   RABBITMQ_PORT: 5552
E2E Tests	Run E2E tests	2026-03-30T15:06:30.2228913Z ##[endgroup]
E2E Tests	Run E2E tests	2026-03-30T15:06:30.3045754Z PHPUnit 10.5.63 by Sebastian Bergmann and contributors.
E2E Tests	Run E2E tests	2026-03-30T15:06:30.3046209Z 
E2E Tests	Run E2E tests	2026-03-30T15:06:30.3046337Z Runtime:       PHP 8.3.30
E2E Tests	Run E2E tests	2026-03-30T15:06:30.3046794Z Configuration: /home/runner/work/rabbit-stream/rabbit-stream/phpunit.xml
E2E Tests	Run E2E tests	2026-03-30T15:06:30.3047103Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.0705928Z EEE..........................................................SSS. 65 / 79 ( 82%)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8243576Z ..............                                                    79 / 79 (100%)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8244135Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8244436Z Time: 01:36.516, Memory: 10.00 MB
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8244736Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8245071Z Access Refused (CrazyGoat\RabbitStream\Tests\E2E\AccessRefused)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8245849Z  ✘ Create stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8246288Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8247210Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8247814Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8248243Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8248859Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8249667Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8250277Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8250889Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8251519Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8252095Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:77
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8252462Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8252694Z  ✘ Create stream access refused has response code
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8252946Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8253520Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8254100Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8254522Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8255167Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8255717Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8256354Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8257181Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8257767Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8258348Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:97
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8258723Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8258909Z  ✘ Delete stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8259465Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8260540Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8261382Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8261839Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8262442Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8263241Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8263826Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8264404Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8264950Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8265514Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:118
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8265884Z    │
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8265964Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8266170Z Amqp Message Decoder E2E (CrazyGoat\RabbitStream\Tests\E2E\AmqpMessageDecoderE2E)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8266584Z  ✔ Decode simple published message
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8266851Z  ✔ Decode message with properties
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8267131Z  ✔ Decode message with application properties
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8267408Z  ✔ Decode multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8267556Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8267666Z Close (CrazyGoat\RabbitStream\Tests\E2E\Close)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8267964Z  ✔ Close command
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8268064Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8268194Z Connection (CrazyGoat\RabbitStream\Tests\E2E\Connection)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8268573Z  ✔ Connection create and stream management
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8268848Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8269904Z  ✔ Get metadata
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8270276Z  ✔ Get stream stats
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8270649Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8270907Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8271194Z Connection Handshake (CrazyGoat\RabbitStream\Tests\E2E\ConnectionHandshake)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8271616Z  ✔ Peer properties exchange
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8271879Z  ✔ Sasl handshake
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8272077Z  ✔ Sasl authenticate
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8272273Z  ✔ Full handshake
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8272491Z  ✔ Invalid credentials throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8272652Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8272857Z Connection Resilience (CrazyGoat\RabbitStream\Tests\E2E\ConnectionResilience)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8273331Z  ✔ Operation after socket disconnect
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8273674Z  ✔ Is connected returns false after socket error
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8274014Z  ✔ No resource leaks after socket error
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8274199Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8274335Z Consumer (CrazyGoat\RabbitStream\Tests\E2E\Consumer)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8274707Z  ✔ Read returns empty on timeout
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8274974Z  ✔ Produce and consume with read
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8275252Z  ✔ Read one returns single message
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8275528Z  ✔ Store and query offset
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8275789Z  ✔ Auto commit on close stores last offset
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8276117Z  ✔ No auto commit on close does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8276530Z  ✔ Auto commit on close with no messages does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8276798Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8276940Z Create Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateStream)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8277281Z  ✔ Create stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8277526Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8277792Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8278183Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8278368Z Create Super Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8278789Z  ✔ Create super stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8279307Z  ✔ Create duplicate super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8279480Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8279695Z Declare Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeclarePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8280111Z  ✔ Declare publisher with reference
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8280412Z  ✔ Declare publisher without reference
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8280738Z  ✔ Declare publisher on non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8280966Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8281129Z Delete Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeletePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8281547Z  ✔ Delete publisher after declare
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8281828Z  ✔ Delete non existent publisher throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8281991Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8282359Z Delete Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteStream)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8282706Z  ✔ Delete stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8282951Z  ✔ Delete non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8283125Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8283322Z Delete Super Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8283707Z  ✔ Delete super stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8283962Z  ✔ Delete non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8284130Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8284372Z Exchange Command Versions (CrazyGoat\RabbitStream\Tests\E2E\ExchangeCommandVersions)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8284843Z  ✔ Exchange command versions
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8284979Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8285239Z Named Producer Deduplication (CrazyGoat\RabbitStream\Tests\E2E\NamedProducerDeduplication)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8285793Z  ✔ Named producer deduplication across reconnect
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8285984Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8286213Z Osiris Chunk Parser E2E (CrazyGoat\RabbitStream\Tests\E2E\OsirisChunkParserE2E)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8286655Z  ✔ Parse real chunk from rabbit m q
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8286960Z  ✔ Chunk parser with single message
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8287118Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8287274Z Partitions (CrazyGoat\RabbitStream\Tests\E2E\Partitions)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8287667Z  ✔ Partitions for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8288017Z  ✔ Partitions returns streams for super stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8288225Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8288360Z Producer (CrazyGoat\RabbitStream\Tests\E2E\Producer)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8288657Z  ✔ Send and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8288895Z  ✔ Send batch and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8289387Z  ✔ Get last publishing id
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8289649Z  ✔ Query sequence for named producer
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8289934Z  ✔ Wait for confirms timeout throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8290215Z  ✔ Producer remains usable after timeout
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8290381Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8290659Z Producer Consumer Offset Resume (CrazyGoat\RabbitStream\Tests\E2E\ProducerConsumerOffsetResume)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8291167Z  ✔ Full lifecycle with offset resume
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8291345Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8291460Z Publish (CrazyGoat\RabbitStream\Tests\E2E\Publish)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8291780Z  ✔ Publish single message
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8292024Z  ✔ Publish multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8292184Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8292351Z Publish Error (CrazyGoat\RabbitStream\Tests\E2E\PublishError)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8292766Z  ✔ Publish error callback is invoked when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8293207Z  ✔ Publish error callback is invoked for batch when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8293450Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8293659Z Query Publisher Sequence (CrazyGoat\RabbitStream\Tests\E2E\QueryPublisherSequence)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8294198Z  ✔ Query publisher sequence returns zero for new publisher
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8294614Z  ✔ Query publisher sequence returns last published id
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8294836Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8294935Z Route (CrazyGoat\RabbitStream\Tests\E2E\Route)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8295308Z  ✔ Route for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8295620Z  ✔ Route returns streams for routing key
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8295968Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8296203Z Store Offset Query Offset E2E (CrazyGoat\RabbitStream\Tests\E2E\StoreOffsetQueryOffsetE2E)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8296668Z  ✔ Store and query offset at protocol level
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8296977Z  ✔ Query offset for non existent reference
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8297301Z  ✔ Store offset multiple times last value wins
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8297497Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8297633Z Stream Stats (CrazyGoat\RabbitStream\Tests\E2E\StreamStats)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8298003Z  ✔ Stream stats returns statistics
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8298336Z  ✔ Stream stats for non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8298538Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8298677Z Subscribe (CrazyGoat\RabbitStream\Tests\E2E\Subscribe)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8299264Z  ✔ Subscribe to stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8299571Z  ✔ Subscribe with offset last
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8300440Z  ✔ Subscribe with offset next
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8300920Z  ✔ Subscribe to non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8301420Z  ✔ Duplicate subscription id throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8301713Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8301964Z Unsubscribe (CrazyGoat\RabbitStream\Tests\E2E\Unsubscribe)
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8302565Z  ✔ Unsubscribe from stream
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8303151Z  ✔ Unsubscribe non existent subscription throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8303562Z  ✔ Unsubscribe already unsubscribed throws
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8304056Z 
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8304214Z ERRORS!
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8304590Z Tests: 79, Assertions: 410, Errors: 3, Skipped: 3.
E2E Tests	Run E2E tests	2026-03-30T15:08:06.8350295Z ##[error]Process completed with exit code 2.
```


```

## CI Pipeline Failed

Pipeline checks failed. Logs saved to `/home/decodo/work/rabbit-stream/.oda/artifacts/152/pipeline-fail.log`.

```
## Failed CI Checks

Failed: E2E Tests

### Run 23752089628

```
E2E Tests	Run E2E tests	﻿2026-03-30T15:12:26.5709526Z ##[group]Run ./vendor/bin/phpunit --testsuite e2e --testdox
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5709956Z [36;1m./vendor/bin/phpunit --testsuite e2e --testdox[0m
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5726607Z shell: /usr/bin/bash -e {0}
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5726845Z env:
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727017Z   COMPOSER_PROCESS_TIMEOUT: 0
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727234Z   COMPOSER_NO_INTERACTION: 1
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727437Z   COMPOSER_NO_AUDIT: 1
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727617Z   RABBITMQ_HOST: 127.0.0.1
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727813Z   RABBITMQ_PORT: 5552
E2E Tests	Run E2E tests	2026-03-30T15:12:26.5727986Z ##[endgroup]
E2E Tests	Run E2E tests	2026-03-30T15:12:26.6444223Z PHPUnit 10.5.63 by Sebastian Bergmann and contributors.
E2E Tests	Run E2E tests	2026-03-30T15:12:26.6444738Z 
E2E Tests	Run E2E tests	2026-03-30T15:12:26.6444905Z Runtime:       PHP 8.3.30
E2E Tests	Run E2E tests	2026-03-30T15:12:26.6445486Z Configuration: /home/runner/work/rabbit-stream/rabbit-stream/phpunit.xml
E2E Tests	Run E2E tests	2026-03-30T15:12:26.6446001Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:02.4277584Z EEE..........................................................SSS. 65 / 79 ( 82%)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1840494Z ..............                                                    79 / 79 (100%)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1841439Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1922167Z Time: 01:36.536, Memory: 10.00 MB
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1925371Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1926061Z Access Refused (CrazyGoat\RabbitStream\Tests\E2E\AccessRefused)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1926912Z  ✘ Create stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1927228Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1928101Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1928914Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1929537Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1930523Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1931658Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1932579Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1933516Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1934369Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1935235Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:77
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1935794Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1936174Z  ✘ Create stream access refused has response code
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1936557Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1937484Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1938370Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1939117Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1940137Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1941069Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1942338Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1943377Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1944729Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1945887Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:97
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1946533Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1946878Z  ✘ Delete stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1947287Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1948319Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1949322Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1949959Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1950926Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1952305Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1953296Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1954366Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1955426Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1956336Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:118
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1956937Z    │
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1957067Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1957388Z Amqp Message Decoder E2E (CrazyGoat\RabbitStream\Tests\E2E\AmqpMessageDecoderE2E)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1958023Z  ✔ Decode simple published message
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1958315Z  ✔ Decode message with properties
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1958613Z  ✔ Decode message with application properties
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1958892Z  ✔ Decode multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1959045Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1959156Z Close (CrazyGoat\RabbitStream\Tests\E2E\Close)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1959416Z  ✔ Close command
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1959530Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1959680Z Connection (CrazyGoat\RabbitStream\Tests\E2E\Connection)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1960011Z  ✔ Connection create and stream management
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1960292Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1960526Z  ✔ Get metadata
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1960713Z  ✔ Get stream stats
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1960937Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1961086Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1961469Z Connection Handshake (CrazyGoat\RabbitStream\Tests\E2E\ConnectionHandshake)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1961862Z  ✔ Peer properties exchange
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1962082Z  ✔ Sasl handshake
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1962290Z  ✔ Sasl authenticate
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1962496Z  ✔ Full handshake
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1962714Z  ✔ Invalid credentials throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1962853Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1963049Z Connection Resilience (CrazyGoat\RabbitStream\Tests\E2E\ConnectionResilience)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1963470Z  ✔ Operation after socket disconnect
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1963771Z  ✔ Is connected returns false after socket error
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1964076Z  ✔ No resource leaks after socket error
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1964236Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1964366Z Consumer (CrazyGoat\RabbitStream\Tests\E2E\Consumer)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1964664Z  ✔ Read returns empty on timeout
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1964927Z  ✔ Produce and consume with read
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1965179Z  ✔ Read one returns single message
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1965437Z  ✔ Store and query offset
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1965689Z  ✔ Auto commit on close stores last offset
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1965983Z  ✔ No auto commit on close does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1966353Z  ✔ Auto commit on close with no messages does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1966570Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1966718Z Create Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateStream)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1967029Z  ✔ Create stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1967241Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1967490Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1967634Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1967980Z Create Super Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1968339Z  ✔ Create super stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1968582Z  ✔ Create duplicate super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1968745Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1968915Z Declare Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeclarePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1969282Z  ✔ Declare publisher with reference
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1969551Z  ✔ Declare publisher without reference
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1969859Z  ✔ Declare publisher on non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1970051Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1970213Z Delete Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeletePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1970571Z  ✔ Delete publisher after declare
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1970842Z  ✔ Delete non existent publisher throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1970998Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1971554Z Delete Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteStream)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1971878Z  ✔ Delete stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1972092Z  ✔ Delete non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1972263Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1972434Z Delete Super Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1972777Z  ✔ Delete super stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1973025Z  ✔ Delete non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1973188Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1973406Z Exchange Command Versions (CrazyGoat\RabbitStream\Tests\E2E\ExchangeCommandVersions)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1973819Z  ✔ Exchange command versions
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1973959Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1974208Z Named Producer Deduplication (CrazyGoat\RabbitStream\Tests\E2E\NamedProducerDeduplication)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1974677Z  ✔ Named producer deduplication across reconnect
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1974867Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1975058Z Osiris Chunk Parser E2E (CrazyGoat\RabbitStream\Tests\E2E\OsirisChunkParserE2E)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1975455Z  ✔ Parse real chunk from rabbit m q
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1975724Z  ✔ Chunk parser with single message
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1975869Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1976025Z Partitions (CrazyGoat\RabbitStream\Tests\E2E\Partitions)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1976380Z  ✔ Partitions for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1976704Z  ✔ Partitions returns streams for super stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1976880Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1976996Z Producer (CrazyGoat\RabbitStream\Tests\E2E\Producer)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1977304Z  ✔ Send and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1977549Z  ✔ Send batch and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1977800Z  ✔ Get last publishing id
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1978046Z  ✔ Query sequence for named producer
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1978306Z  ✔ Wait for confirms timeout throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1978579Z  ✔ Producer remains usable after timeout
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1978736Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1978986Z Producer Consumer Offset Resume (CrazyGoat\RabbitStream\Tests\E2E\ProducerConsumerOffsetResume)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1979453Z  ✔ Full lifecycle with offset resume
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1979605Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1979712Z Publish (CrazyGoat\RabbitStream\Tests\E2E\Publish)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1980008Z  ✔ Publish single message
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1980251Z  ✔ Publish multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1980390Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1980530Z Publish Error (CrazyGoat\RabbitStream\Tests\E2E\PublishError)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1980926Z  ✔ Publish error callback is invoked when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1981567Z  ✔ Publish error callback is invoked for batch when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1981808Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1982016Z Query Publisher Sequence (CrazyGoat\RabbitStream\Tests\E2E\QueryPublisherSequence)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1982482Z  ✔ Query publisher sequence returns zero for new publisher
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1982854Z  ✔ Query publisher sequence returns last published id
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1983048Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1983168Z Route (CrazyGoat\RabbitStream\Tests\E2E\Route)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1983485Z  ✔ Route for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1983782Z  ✔ Route returns streams for routing key
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1984062Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1984286Z Store Offset Query Offset E2E (CrazyGoat\RabbitStream\Tests\E2E\StoreOffsetQueryOffsetE2E)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1984734Z  ✔ Store and query offset at protocol level
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1985017Z  ✔ Query offset for non existent reference
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1985323Z  ✔ Store offset multiple times last value wins
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1985504Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1985645Z Stream Stats (CrazyGoat\RabbitStream\Tests\E2E\StreamStats)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1985972Z  ✔ Stream stats returns statistics
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1986255Z  ✔ Stream stats for non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1986426Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1986558Z Subscribe (CrazyGoat\RabbitStream\Tests\E2E\Subscribe)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1986907Z  ✔ Subscribe to stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1987128Z  ✔ Subscribe with offset last
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1987497Z  ✔ Subscribe with offset next
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1987748Z  ✔ Subscribe to non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1988030Z  ✔ Duplicate subscription id throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1988187Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1988323Z Unsubscribe (CrazyGoat\RabbitStream\Tests\E2E\Unsubscribe)
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1988648Z  ✔ Unsubscribe from stream
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1988920Z  ✔ Unsubscribe non existent subscription throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1989216Z  ✔ Unsubscribe already unsubscribed throws
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1989408Z 
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1989501Z ERRORS!
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1989684Z Tests: 79, Assertions: 410, Errors: 3, Skipped: 3.
E2E Tests	Run E2E tests	2026-03-30T15:14:03.1998485Z ##[error]Process completed with exit code 2.
```


```

## CI Pipeline Failed

Pipeline checks failed. Logs saved to `/home/decodo/work/rabbit-stream/.oda/artifacts/152/pipeline-fail.log`.

```
## Failed CI Checks

Failed: E2E Tests

### Run 23752360044

```
E2E Tests	Run E2E tests	﻿2026-03-30T15:18:06.2149945Z ##[group]Run ./vendor/bin/phpunit --testsuite e2e --testdox
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2150388Z [36;1m./vendor/bin/phpunit --testsuite e2e --testdox[0m
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2176304Z shell: /usr/bin/bash -e {0}
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2176550Z env:
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2176722Z   COMPOSER_PROCESS_TIMEOUT: 0
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2176957Z   COMPOSER_NO_INTERACTION: 1
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2177172Z   COMPOSER_NO_AUDIT: 1
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2177366Z   RABBITMQ_HOST: 127.0.0.1
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2177572Z   RABBITMQ_PORT: 5552
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2177756Z ##[endgroup]
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2983763Z PHPUnit 10.5.63 by Sebastian Bergmann and contributors.
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2984312Z 
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2984737Z Runtime:       PHP 8.3.30
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2985303Z Configuration: /home/runner/work/rabbit-stream/rabbit-stream/phpunit.xml
E2E Tests	Run E2E tests	2026-03-30T15:18:06.2985689Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.0427011Z EEE..........................................................SSS. 65 / 79 ( 82%)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7973634Z ..............                                                    79 / 79 (100%)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7974115Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7974285Z Time: 01:36.496, Memory: 10.00 MB
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7974767Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7975069Z Access Refused (CrazyGoat\RabbitStream\Tests\E2E\AccessRefused)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7975893Z  ✘ Create stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7976300Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7977196Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7998503Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.7999469Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8000650Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8001668Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8002774Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8003951Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8005322Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8006543Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:77
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8007347Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8007861Z  ✘ Create stream access refused has response code
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8008428Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8009677Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8010946Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8011849Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8013191Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8014307Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8015919Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8017216Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8018807Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8020087Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:97
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8020890Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8021286Z  ✘ Delete stream access refused
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8021708Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8022873Z    │ CrazyGoat\RabbitStream\Exception\ProtocolException: Unexpected response code: 0x0008 (AUTHENTICATION_FAILURE: Authentication failure)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8024019Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8025146Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Response/SimpleCorrelatedResponseV1.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8026543Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:85
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8028121Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/ResponseBuilder.php:51
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8029475Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/Serializer/PhpBinarySerializer.php:27
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8030672Z    │ /home/runner/work/rabbit-stream/rabbit-stream/src/StreamConnection.php:293
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8031880Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:59
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8033180Z    │ /home/runner/work/rabbit-stream/rabbit-stream/tests/E2E/AccessRefusedTest.php:118
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8034020Z    │
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8034210Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8034868Z Amqp Message Decoder E2E (CrazyGoat\RabbitStream\Tests\E2E\AmqpMessageDecoderE2E)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8035749Z  ✔ Decode simple published message
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8036351Z  ✔ Decode message with properties
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8037025Z  ✔ Decode message with application properties
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8037762Z  ✔ Decode multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8038113Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8038346Z Close (CrazyGoat\RabbitStream\Tests\E2E\Close)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8039002Z  ✔ Close command
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8039252Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8039563Z Connection (CrazyGoat\RabbitStream\Tests\E2E\Connection)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8040268Z  ✔ Connection create and stream management
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8040829Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8041304Z  ✔ Get metadata
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8041684Z  ✔ Get stream stats
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8042173Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8042485Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8042836Z Connection Handshake (CrazyGoat\RabbitStream\Tests\E2E\ConnectionHandshake)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8043621Z  ✔ Peer properties exchange
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8044115Z  ✔ Sasl handshake
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8044809Z  ✔ Sasl authenticate
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8045281Z  ✔ Full handshake
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8045787Z  ✔ Invalid credentials throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8046088Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8046530Z Connection Resilience (CrazyGoat\RabbitStream\Tests\E2E\ConnectionResilience)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8047514Z  ✔ Operation after socket disconnect
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8048216Z  ✔ Is connected returns false after socket error
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8048971Z  ✔ No resource leaks after socket error
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8049339Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8049600Z Consumer (CrazyGoat\RabbitStream\Tests\E2E\Consumer)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8050349Z  ✔ Read returns empty on timeout
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8050997Z  ✔ Produce and consume with read
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8051594Z  ✔ Read one returns single message
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8052115Z  ✔ Store and query offset
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8052601Z  ✔ Auto commit on close stores last offset
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8052942Z  ✔ No auto commit on close does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8053334Z  ✔ Auto commit on close with no messages does not store offset
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8053580Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8053735Z Create Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateStream)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8054069Z  ✔ Create stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8054301Z  ✔ Create stream with arguments
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8054867Z  ✔ Create duplicate stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8055055Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8055455Z Create Super Stream (CrazyGoat\RabbitStream\Tests\E2E\CreateSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8055834Z  ✔ Create super stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8056078Z  ✔ Create duplicate super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8056247Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8056423Z Declare Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeclarePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8056807Z  ✔ Declare publisher with reference
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8057079Z  ✔ Declare publisher without reference
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8057391Z  ✔ Declare publisher on non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8057589Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8057758Z Delete Publisher (CrazyGoat\RabbitStream\Tests\E2E\DeletePublisher)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8058120Z  ✔ Delete publisher after declare
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8058391Z  ✔ Delete non existent publisher throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8058557Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8058843Z Delete Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteStream)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8059166Z  ✔ Delete stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8059380Z  ✔ Delete non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8059556Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8059732Z Delete Super Stream (CrazyGoat\RabbitStream\Tests\E2E\DeleteSuperStream)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8060102Z  ✔ Delete super stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8060344Z  ✔ Delete non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8060515Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8060740Z Exchange Command Versions (CrazyGoat\RabbitStream\Tests\E2E\ExchangeCommandVersions)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8061164Z  ✔ Exchange command versions
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8061306Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8061555Z Named Producer Deduplication (CrazyGoat\RabbitStream\Tests\E2E\NamedProducerDeduplication)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8062042Z  ✔ Named producer deduplication across reconnect
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8062239Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8062436Z Osiris Chunk Parser E2E (CrazyGoat\RabbitStream\Tests\E2E\OsirisChunkParserE2E)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8062861Z  ✔ Parse real chunk from rabbit m q
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8063126Z  ✔ Chunk parser with single message
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8063280Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8063428Z Partitions (CrazyGoat\RabbitStream\Tests\E2E\Partitions)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8063794Z  ✔ Partitions for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8064128Z  ✔ Partitions returns streams for super stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8064309Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8064661Z Producer (CrazyGoat\RabbitStream\Tests\E2E\Producer)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8064997Z  ✔ Send and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8065249Z  ✔ Send batch and wait for confirms
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8065497Z  ✔ Get last publishing id
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8065740Z  ✔ Query sequence for named producer
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8066004Z  ✔ Wait for confirms timeout throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8066275Z  ✔ Producer remains usable after timeout
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8066439Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8066706Z Producer Consumer Offset Resume (CrazyGoat\RabbitStream\Tests\E2E\ProducerConsumerOffsetResume)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8067193Z  ✔ Full lifecycle with offset resume
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8067347Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8067464Z Publish (CrazyGoat\RabbitStream\Tests\E2E\Publish)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8067800Z  ✔ Publish single message
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8068038Z  ✔ Publish multiple messages
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8068176Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8068321Z Publish Error (CrazyGoat\RabbitStream\Tests\E2E\PublishError)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8068727Z  ✔ Publish error callback is invoked when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8069130Z  ✔ Publish error callback is invoked for batch when stream deleted
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8069378Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8069595Z Query Publisher Sequence (CrazyGoat\RabbitStream\Tests\E2E\QueryPublisherSequence)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8070077Z  ✔ Query publisher sequence returns zero for new publisher
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8070448Z  ✔ Query publisher sequence returns last published id
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8070652Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8070757Z Route (CrazyGoat\RabbitStream\Tests\E2E\Route)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8071077Z  ✔ Route for non existent super stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8071376Z  ✔ Route returns streams for routing key
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8071676Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8072050Z Store Offset Query Offset E2E (CrazyGoat\RabbitStream\Tests\E2E\StoreOffsetQueryOffsetE2E)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8072828Z  ✔ Store and query offset at protocol level
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8073340Z  ✔ Query offset for non existent reference
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8073921Z  ✔ Store offset multiple times last value wins
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8074228Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8074654Z Stream Stats (CrazyGoat\RabbitStream\Tests\E2E\StreamStats)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8075261Z  ✔ Stream stats returns statistics
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8075831Z  ✔ Stream stats for non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8076137Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8076356Z Subscribe (CrazyGoat\RabbitStream\Tests\E2E\Subscribe)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8076969Z  ✔ Subscribe to stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8077404Z  ✔ Subscribe with offset last
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8078143Z  ✔ Subscribe with offset next
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8078645Z  ✔ Subscribe to non existent stream throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8079194Z  ✔ Duplicate subscription id throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8079498Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8079647Z Unsubscribe (CrazyGoat\RabbitStream\Tests\E2E\Unsubscribe)
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8079986Z  ✔ Unsubscribe from stream
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8080257Z  ✔ Unsubscribe non existent subscription throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8080576Z  ✔ Unsubscribe already unsubscribed throws
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8081036Z 
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8081178Z ERRORS!
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8081531Z Tests: 79, Assertions: 410, Errors: 3, Skipped: 3.
E2E Tests	Run E2E tests	2026-03-30T15:19:42.8090376Z ##[error]Process completed with exit code 2.
```


```

--- a/tests/E2E/AccessRefusedE2ETest.php
+++ b/tests/E2E/AccessRefusedE2ETest.php
@@ -1,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CrazyGoat\RabbitStream\Tests\E2E;
+
+use CrazyGoat\RabbitStream\Enum\ResponseCodeEnum;
+use CrazyGoat\RabbitStream\Exception\ProtocolException;
+
+/**
+ * @group destructive
+ */
+class AccessRefusedE2ETest extends E2ETestCase
+{
+    private const string RESTRICTED_USER = 'restricted-user';
+    private const string RESTRICTED_PASS = 'restricted-pass';
+    private const string RESTRICTED_VHOST = 'restricted-vhost';
+
+    private static int $managementPort = 15672;
+    private static bool $managementAvailable = false;
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        self::$managementPort = (int)(getenv('RABBITMQ_MANAGEMENT_PORT') ?: self::$managementPort);
+        self::ensureManagementFixtures();
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        self::deleteManagement('/users/' . self::RESTRICTED_USER);
+        self::deleteManagement('/vhosts/' . self::RESTRICTED_VHOST);
+        parent::tearDownAfterClass();
+    }
+
+    private static function ensureManagementFixtures(): void
+    {
+        if (!self::putManagement('/vhosts/' . self::RESTRICTED_VHOST)) {
+            return;
+        }
+
+        if (
+            !self::putManagement('/users/' . self::RESTRICTED_USER, [
+            'password' => self::RESTRICTED_PASS,
+            'tags' => 'none',
+            ])
+        ) {
+            self::deleteManagement('/vhosts/' . self::RESTRICTED_VHOST);
+            return;
+        }
+
+        $permPath = '/permissions/' . self::RESTRICTED_VHOST . '/' . self::RESTRICTED_USER;
+        if (
+            !self::putManagement($permPath, [
+            'configure' => '',
+            'write' => '.*',
+            'read' => '.*',
+            ])
+        ) {
+            self::deleteManagement('/users/' . self::RESTRICTED_USER);
+            self::deleteManagement('/vhosts/' . self::RESTRICTED_VHOST);
+            return;
+        }
+
+        self::$managementAvailable = true;
+    }
+
+    /** @param array<string, mixed>|null $body */
+    private static function managementRequest(string $method, string $path, ?array $body = null): bool
+    {
+        $url = sprintf('http://%s:%d/api%s', self::$host, self::$managementPort, $path);
+        $headers = [
+            'Authorization: Basic ' . base64_encode('guest:guest'),
+            'Content-Type: application/json',
+        ];
+
+        $http = [
+            'method' => $method,
+            'header' => implode("\r\n", $headers),
+            'ignore_errors' => true,
+        ];
+
+        if ($body !== null) {
+            $http['content'] = json_encode($body, JSON_THROW_ON_ERROR);
+        }
+
+        $context = stream_context_create(['http' => $http]);
+        $result = @file_get_contents($url, false, $context);
+
+        if ($result === false) {
+            return false;
+        }
+
+        $statusLine = $http_response_header[0] ?? '';
+        if ($statusLine !== '' && preg_match('#^HTTP/\d\.\d\s+(\d+)#', $statusLine, $m)) {
+            return (int)$m[1] < 400;
+        }
+
+        return true;
+    }
+
+    private static function deleteManagement(string $path): void
+    {
+        self::managementRequest('DELETE', $path);
+    }
+
+    /** @param array<string, mixed>|null $body */
+    private static function putManagement(string $path, ?array $body = null): bool
+    {
+        return self::managementRequest('PUT', $path, $body);
+    }
+
+    public function testAccessRefusedForCreateWithoutConfigurePermission(): void
+    {
+        if (!self::$managementAvailable) {
+            $this->markTestSkipped('Management API not available');
+        }
+
+        $connection = $this->createConnection(
+            user: self::RESTRICTED_USER,
+            password: self::RESTRICTED_PASS,
+            vhost: self::RESTRICTED_VHOST,
+        );
+
+        $streamName = 'test-forbidden-stream-' . uniqid();
+
+        try {
+            $connection->createStream($streamName);
+            $this->fail('Expected ProtocolException to be thrown');
+        } catch (ProtocolException $e) {
+            $this->assertSame(
+                ResponseCodeEnum::ACCESS_REFUSED,
+                $e->getResponseCode(),
+                'Expected ACCESS_REFUSED response code',
+            );
+            $this->assertStringContainsString(
+                'ACCESS_REFUSED',
+                $e->getMessage(),
+                'Error message should include the response code name',
+            );
+        } finally {
+            $connection->close();
+        }
+    }
+}

